warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -277,27 +277,9 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
         pruneIvaratorCacheDirs();
     }
     
-    // this method will prune any ivarator cache directories that are not available on this node
-    private void pruneIvaratorCacheDirs() throws IOException {
-        ivaratorCacheDirConfigs.removeIf(this::pruneIvaratorCacheDir);
-    }
-    
-    private boolean pruneIvaratorCacheDir(IvaratorCacheDirConfig config) {
-        boolean fsExists = false;
-        
-        // first, make sure the cache configuration is valid
-        if (config.isValid()) {
-            Path basePath = new Path(config.getBasePathURI());
-            
-            try {
-                FileSystem fs = this.getFileSystemCache().getFileSystem(basePath.toUri());
-                fsExists = fs.mkdirs(basePath) || fs.exists(basePath);
-            } catch (Exception e) {
-                log.debug("Ivarator Cache Dir does not exist: " + basePath);
-            }
-        }
-        
-        return !fsExists;
+    // this method will prune any ivarator cache directories that do not have a valid configuration.
+    private void pruneIvaratorCacheDirs() {
+        ivaratorCacheDirConfigs.removeIf(config -> !config.isValid());
```

Comment:
Is there anyway to verify a volume is writable without actually writing to it?

Reply:
FileSystem.access or FileSystem.getAclStatus perhaps

Reply:
Also once we find the first cache directory that is valid and writable, there is no need to check the other cache directories.  This way we can avoid excessive HDFS namenode calls which was the original problem that produced this ticket.  I apologize that these extra requirements were not stated up front.  I had not thought about the problem enough at the time.

Reply:
I will try figuring something out to test if a volume is writable.

With regards to your latest comment, just to make sure I understand, do you want the remaining entries in ivaratorCacheDirConfigs removed after we find the first valid and writable cache directory>

Reply:
@ivakegg can it be assumed that if the base path of a config doesn't exist, its parent does, or its parent is root?

Reply:
I think we can simply prune out the remaining ones once we find one that is valid and writable.
I am not sure we can assume the parent path exists.  The first ancestor that exists must be writable to allow this to work.